pkg/controller/cloud/node_lifecycle_controller_test.go

```diff
@@ -0,0 +1,424 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloud
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog"
+	fakecloud "k8s.io/kubernetes/pkg/cloudprovider/providers/fake"
+	"k8s.io/kubernetes/pkg/controller/testutil"
+)
+
+func Test_NodesDeleted(t *testing.T) {
+	testcases := []struct {
+		name        string
+		fnh         *testutil.FakeNodeHandler
+		fakeCloud   *fakecloud.FakeCloud
+		deleteNodes []*v1.Node
+	}{
+		{
+			"node is not ready and does not exist",
+			&testutil.FakeNodeHandler{
```

Comment:
Please use the field names, these struct literals are fairly large and difficult to read without the names included.